index.js

```diff
@@ -33,6 +34,20 @@ function logid (pkg, stage) {
   return pkg._id + '~' + stage + ':'
 }
 
+function hookExists (dir, stage, cb) {
+  const hook = path.join(dir, '.hooks', stage)
+  const cachedStatError = hookStatCache.get(hook)
+
+  if (cachedStatError === undefined) {
+    return fs.stat(hook, function (statError) {
+      hookStatCache.set(hook, statError)
+      cb(statError)
+    })
+  }
+
+  return cb(cachedStatError)
```

Comment:
could you [dezalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony) this? (just `setImmediate` the callback)